lib/radar_client.js

```diff
@@ -130,210 +132,184 @@ Client.prototype.stream = function(scope) {
+                  this._configuration.userId, this._configuration.userType);
 
-  return this._write(message, callback);
+  return this._write(request, callback);
 };
 
 Client.prototype.publish = function(scope, value, callback) {
-  return this._write({
-    op: 'publish',
-    to: scope,
-    value: value
-  }, callback);
+  var request = Request.buildPublish(scope, value);
+  return this._write(request, callback);
 };
 
 Client.prototype.subscribe = function(scope, options, callback) {
-  var message = { op: 'subscribe', to: scope };
-  if (typeof options == 'function') {
-    callback = options;
-  } else {
-    message.options = options;
-  }
-  return this._write(message, callback);
+  callback = _chooseFunction(options, callback);
+  options = _nullIfFunction(options);
+
+  var request = Request.buildSubscribe(scope, options);
+
+  return this._write(request, callback);
 };
 
 Client.prototype.unsubscribe = function(scope, callback) {
-  return this._write({ op: 'unsubscribe', to: scope }, callback);
+  var request = Request.buildUnsubscribe(scope);
+  return this._write(request, callback);
 };
 
-var v2ResponseToV1 = function (message) {
-  // Translate v2 response to v1
-  var value = {}, userId;
-  for (userId in message.value) {
-    if (message.value.hasOwnProperty(userId)) {
-      // Skip when not defined; causes exception in FF for 'Work Offline'
-      if (!message.value[userId]) { continue; }
-      value[userId] = message.value[userId].userType;
-    }
-  }
-  message.value = value;
-  message.op = 'online';
+// sync returns the actual value of the operation
+Client.prototype.sync = function (scope, options, callback) {
+  var request, onResponse, v1Presence;
 
-  return message;
-};
+  callback = _chooseFunction(options, callback);
+  options = _nullIfFunction(options);
 
-Client.prototype.sync = function (scope, options, callback) {
-  var message = { op: 'sync', to: scope };
-  // options is an optional argument
-  if (typeof options == 'function') {
-    callback = options;
-  } else {
-    message.options = options;
-  }
-  // So force v2 and translate the result to v1 format.
-  var v1Presence = !message.options && scope.match(/^presence.+/);
-  var onResponse = function (message) {
-    if (message && message.to && message.to === scope) {
+  request = Request.buildSync(scope, options);
+
+  //v1Presence = request.getVersion() === 1 && request.isPresence();
+  v1Presence = !options && request.isPresence();
+  onResponse = function (response) {
+    if (response && response.isFor(request)) {
       if (v1Presence) {
-        message = v2ResponseToV1(message);
+        response.forceV1Response();
       }
-
       if (callback) {
-        callback(message);
+        callback(response.getMessage());
       }
       return true;
     }
     return false;
   };
 
-  if (v1Presence) {
-    message.options = { version: 2 };
-  }
-
+  if (v1Presence) { request.setAttr('options', { version: 2 }); }
```

Comment:
Actually move this part into buildSync(). There is no v1Presence option anymore for anyone using request.js. The kludgy code is only here.